ci/release/update-version.sh

```diff
@@ -60,19 +60,15 @@ sed_runner 's/release = .*/release = '"'${NEXT_FULL_TAG}'"'/g' docs/source/conf.
 sed_runner "s/\(PROJECT_NUMBER.*=\).*/\1 \"${NEXT_SHORT_TAG}\"/g" cpp/Doxyfile.in
 
 for FILE in dependencies.yaml conda/environments/*.yaml; do
-   sed_runner "s/cudf=${CURRENT_SHORT_TAG}/cudf=${NEXT_SHORT_TAG}/g" ${FILE};
-   sed_runner "s/rmm=${CURRENT_SHORT_TAG}/rmm=${NEXT_SHORT_TAG}/g" ${FILE};
-   sed_runner "s/dask-cuda=${CURRENT_SHORT_TAG}/dask-cuda=${NEXT_SHORT_TAG}/g" ${FILE};
-   sed_runner "s/dask-cudf=${CURRENT_SHORT_TAG}/dask-cudf=${NEXT_SHORT_TAG}/g" ${FILE};
-   sed_runner "s/libcumlprims=${CURRENT_SHORT_TAG}/libcumlprims=${NEXT_SHORT_TAG}/g" ${FILE};
-   sed_runner "s/libraft-headers=${CURRENT_SHORT_TAG}/libraft-headers=${NEXT_SHORT_TAG}/g" ${FILE};
-   sed_runner "s/libraft-distance=${CURRENT_SHORT_TAG}/libraft-distance=${NEXT_SHORT_TAG}/g" ${FILE};
-   sed_runner "s/libraft-nn=${CURRENT_SHORT_TAG}/libraft-nn=${NEXT_SHORT_TAG}/g" ${FILE};
-   sed_runner "s/raft-dask=${CURRENT_SHORT_TAG}/raft-dask=${NEXT_SHORT_TAG}/g" ${FILE};
-   sed_runner "s/pylibraft=${CURRENT_SHORT_TAG}/pylibraft=${NEXT_SHORT_TAG}/g" ${FILE};
-   sed_runner "s/rapids-build-env=${CURRENT_SHORT_TAG}/rapids-build-env=${NEXT_SHORT_TAG}/g" ${FILE};
-   sed_runner "s/rapids-notebook-env=${CURRENT_SHORT_TAG}/rapids-notebook-env=${NEXT_SHORT_TAG}/g" ${FILE};
-   sed_runner "s/rapids-doc-env=${CURRENT_SHORT_TAG}/rapids-doc-env=${NEXT_SHORT_TAG}/g" ${FILE};
+   sed_runner "s/rmm==.*\",/rmm==${NEXT_SHORT_TAG_PEP440}.*\",/g" ${FILE};
+   sed_runner "s/cudf==.*\",/cudf==${NEXT_SHORT_TAG_PEP440}.*\",/g" ${FILE};
+   sed_runner "s/dask-cuda==.*\",/dask-cuda==${NEXT_SHORT_TAG_PEP440}.*\",/g" ${FILE};
+   sed_runner "s/dask-cudf==.*\",/dask-cudf==${NEXT_SHORT_TAG_PEP440}.*\",/g" ${FILE};
+   sed_runner "s/libcumlprims==.*\",/libcumlprims==${NEXT_SHORT_TAG_PEP440}.*\",/g" ${FILE};
+   sed_runner "s/libraft-headers==.*\",/libraft-headers==${NEXT_SHORT_TAG_PEP440}.*\",/g" ${FILE};
+   sed_runner "s/libraft==.*\",/libraft==${NEXT_SHORT_TAG_PEP440}.*\",/g" ${FILE};
+   sed_runner "s/raft-dask==.*\",/raft-dask==${NEXT_SHORT_TAG_PEP440}.*\",/g" ${FILE};
+   sed_runner "s/pylibraft==.*\",/pylibraft==${NEXT_SHORT_TAG_PEP440}.*\",/g" ${FILE};
```

Comment:
```suggestion
   sed_runner "/- cudf==/            s/==.*/==${NEXT_SHORT_TAG_PEP440}\.*/g" ${FILE};
   sed_runner "/- dask-cuda==/       s/==.*/==${NEXT_SHORT_TAG_PEP440}\.*/g" ${FILE};
   sed_runner "/- dask-cudf==/       s/==.*/==${NEXT_SHORT_TAG_PEP440}\.*/g" ${FILE};
   sed_runner "/- libcumlprims==/    s/==.*/==${NEXT_SHORT_TAG_PEP440}\.*/g" ${FILE};
   sed_runner "/- libraft-headers==/ s/==.*/==${NEXT_SHORT_TAG_PEP440}\.*/g" ${FILE};
   sed_runner "/- libraft==/         s/==.*/==${NEXT_SHORT_TAG_PEP440}\.*/g" ${FILE};
   sed_runner "/- librmm==/          s/==.*/==${NEXT_SHORT_TAG_PEP440}\.*/g" ${FILE};
   sed_runner "/- pylibraft==/       s/==.*/==${NEXT_SHORT_TAG_PEP440}\.*/g" ${FILE};
   sed_runner "/- raft-dask==/       s/==.*/==${NEXT_SHORT_TAG_PEP440}\.*/g" ${FILE};
   sed_runner "/- rmm==/             s/==.*/==${NEXT_SHORT_TAG_PEP440}\.*/g" ${FILE};
```

This suggestion cleans up the sed commands in the following ways:

- only specifies the dependency name once
- explicitly states each dependency name, eliminating ambiguity between dependency names like `cudf`/`dask-cudf` and `rmm`/`librmm` 

This could be deduped even further with the following suggestion, which I can't suggest via GitHub since it touches lines that you haven't edited:


```sh
DEPENDENCIES=(
  cudf
  dask-cuda
  dask-cudf
  libcumlprims
  libraft-headers
  libraft
  librmm
  pylibraft
  raft-dask
  rmm
)
for FILE in dependencies.yaml conda/environments/*.yaml; do
  for DEP in "${DEPENDENCIES[@]}"; do
    sed_runner "/- ${DEP}==/ s/==.*/==${NEXT_SHORT_TAG_PEP440}\.*/g" ${FILE};
  done
done
```


These changes can be tested locally by running commands like:

```
bash ci/release/update-version.sh 23.10.00
bash ci/release/update-version.sh 23.12.00
bash ci/release/update-version.sh 23.10.00
```

Reply:
Thanks @ajschmidt8. I applied your better suggestion and confirmed it works locally. Updated in 50af9b6ab75328a8018a5859dfd4281f135541a1.